src/ferui-components/tree-view/interfaces.ts

```diff
@@ -0,0 +1,108 @@
+import { TemplateRef } from '@angular/core';
+
+/**
+ * Tree Node Data Interface
+ */
+export interface TreeNodeData<T> {
+  data: T;
+  label: string;
+}
+
+/**
+ * Paging Params Interface
+ */
+export interface PagingParams {
+  offset: number;
+  limit: number;
+}
+
+/**
+ * Tree Node Data Retriever Interface for client-side tree view
+ */
+export interface TreeNodeDataRetriever<T> {
+  // Get promise that returns TreeNodeData children array from Tree Node
+  getChildNodeData(parent: TreeNode<T>): Promise<Array<TreeNodeData<T>>>;
+  // Get promise that returns whether TreeNode has children nodes
+  hasChildNodes(parent: TreeNode<T>): Promise<boolean>;
+  // based on the current TreeNode and its status, we allow the developer to give us an icon template
+  getIconTemplate?(treeNode: TreeNode<T>): TemplateRef<any>;
+  // If developer wishes to give us a template ref we shall render this template in the view
+  getNodeTemplate?(): TemplateRef<any>;
+}
+
+/**
+ * Paged Tree Node Data Retriever for server-side tree view
+ */
+export interface PagedTreeNodeDataRetriever<T> extends TreeNodeDataRetriever<T> {
+  getPagedChildNodeData(parent: TreeNode<T>, pagingParams: PagingParams): Promise<Array<TreeNodeData<T>>>;
```

Comment:
I think this should take `TreeNodeData<T>`

Reply:
Fixed all these